views/Login.js

```diff
@@ -1,8 +1,147 @@
-import React from 'react';
-import View from 'react-native';
+import React, { useContext, useEffect } from 'react';
+import {
+  TextInput,
+  SafeAreaView,
+  View,
+  StyleSheet,
+  Button,
+  Text,
+} from 'react-native';
+import Typography from '../components/Typography';
+import { theme } from '../themes';
+import { Formik } from 'formik';
+import { useUser, useLogin } from '../hooks/ApiHooks';
+import { MainContext } from '../context/MainContext';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import * as yup from 'yup';
+import PropTypes from 'prop-types';
+import { Link } from '@react-navigation/native';
 
-const Login = () => {
-  return <View />;
+const LoginSchema = yup.object({
+  username: yup.string().required('Username required'),
+  password: yup.string().required('Password required'),
+});
+
+const Login = ({ navigation }) => {
+  const { getUserByToken } = useUser();
+  const { setIsLoggedIn, setUser } = useContext(MainContext);
+  const { postLogin } = useLogin();
+
+  const checkToken = async () => {
+    const userToken = await AsyncStorage.getItem('userToken');
+    if (!userToken) {
+      return;
+    }
+    try {
+      const userData = await getUserByToken(userToken);
+      console.log(userData);
+      setUser(userData);
+      setIsLoggedIn(true);
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  const onSubmit = async (data) => {
+    try {
+      const userData = await postLogin(data);
+      await AsyncStorage.setItem('userToken', userData.token);
+      setUser(userData.user);
+      setIsLoggedIn(true);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  useEffect(() => {
+    checkToken();
+  }, []);
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <View style={styles.loginContainer}>
+        <Typography variant="h2" text="Login" color={theme.colors.primary} />
+        <Formik
+          validationSchema={LoginSchema}
+          initialValues={{ username: '', password: '' }}
+          onSubmit={(values) => {
+            onSubmit(values);
+          }}
+        >
+          {({
+            handleChange,
+            handleBlur,
+            handleSubmit,
+            values,
+            errors,
+            touched,
+          }) => (
+            <View>
+              <TextInput
```

Comment:
Can you set the text input to be not auto-capitalize? Because it will provide the wrong username